modules/integration/tests-common/admin-clients/src/main/java/org/wso2/mb/integration/common/clients/AndesClient.java

```diff
@@ -1,589 +1,317 @@
+        log.info("TPS:" + this.getConsumerTPS() + " AverageLatency:" + this.getAverageLatency());
+    }
 
-            for (QueueMessageReceiver qListener : queueListeners) {
-                qListener.stopListening();
-            }
+    /**
+     * Gets the received messages for all consumers in the client.
+     *
+     * @return The total number of messages received for all consumers.
+     */
+    public long getReceivedMessageCount() {
+        long allReceivedMessageCount = 0L;
+        for (AndesJMSConsumer consumer : consumers) {
+            allReceivedMessageCount = allReceivedMessageCount + consumer.getReceivedMessageCount();
         }
-
+        return allReceivedMessageCount;
     }
 
-    public int getReceivedqueueMessagecount() {
-        return queueMessageCounter.get();
+    /**
+     * Gets the average transactions per second for consumer(s).
+     *
+     * @return The average TPS.
+     */
+    public double getConsumerTPS() {
+        double tps = 0L;
+        for (AndesJMSConsumer consumer : consumers) {
+            tps = tps + consumer.getConsumerTPS();
+        }
+        return tps / consumers.size();
     }
 
-    public int getReceivedTopicMessagecount() {
-        return topicMessageCounter.get();
+    /**
+     * Gets the average latency for consumer(s).
+     *
+     * @return The average latency.
+     */
+    public double getAverageLatency() {
+        double averageLatency = 0L;
+        for (AndesJMSConsumer consumer : consumers) {
+            averageLatency = averageLatency + consumer.getAverageLatency();
+        }
+        return averageLatency / consumers.size();
     }
 
-    public Map<Long, Integer> checkIfMessagesAreDuplicated() {
-        org.wso2.mb.integration.common.clients.operations.utils.AndesClientUtils.flushPrintWriter();
-        AndesClientOutputParser andesClientOutputParser = new AndesClientOutputParser(filePathToWriteReceivedMessages);
-        return andesClientOutputParser.checkIfMessagesAreDuplicated();
+    /**
+     * Gets the number of messages sent by the publisher(s).
+     *
+     * @return The number of messages.
+     */
+    public long getSentMessageCount() {
+        long allSentMessageCount = 0L;
+        for (AndesJMSPublisher publisher : publishers) {
+            allSentMessageCount = allSentMessageCount + publisher.getSentMessageCount();
+        }
+        return allSentMessageCount;
     }
 
-    public boolean checkIfMessagesAreInOrder() {
-        AndesClientOutputParser andesClientOutputParser = new AndesClientOutputParser(filePathToWriteReceivedMessages);
-        return andesClientOutputParser.checkIfMessagesAreInOrder();
+    /**
+     * Gets the average transactions per seconds for publisher(s). Suppressing "UnusedDeclaration"
+     * as the client acts as an service.
+     *
+     * @return the average transactions per seconds.
+     */
+    @SuppressWarnings("UnusedDeclaration")
+    public double getPublisherTPS() {
```

Comment:
See https://github.com/hemikak/product-mb/commit/a5f87403f7cb36b65e28c25f4a6aef12655d2b45 for comment.